azurerm/internal/services/apimanagement/api_management_api_diagnostic_resource.go

```diff
@@ -0,0 +1,169 @@
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"identifier": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"applicationinsights",
+					"azuremonitor",
+				}, false),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"api_management_name": azure.SchemaApiManagementName(),
+
+			"api_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: helpersvalidate.ApiManagementApiName,
+			},
+
+			"api_management_logger_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ApiManagementLoggerID,
+			},
+
+			"enabled": {
+				Type:       schema.TypeBool,
+				Optional:   true,
+				Deprecated: "this property has been removed from the API and will be removed in version 3.0 of the provider",
+			},
+		},
+	}
+}
+
+func resourceArmApiManagementApiDiagnosticCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.ApiDiagnosticClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	diagnosticId := d.Get("identifier").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("api_management_name").(string)
+	apiName := d.Get("api_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, serviceName, apiName, diagnosticId)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Diagnostic %q (Resource Group %q / API Management Service %q / API %q): %s", diagnosticId, resourceGroup, serviceName, apiName, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_api_management_api_diagnostic", *existing.ID)
+		}
+	}
+
+	parameters := apimanagement.DiagnosticContract{
+		DiagnosticContractProperties: &apimanagement.DiagnosticContractProperties{
+			LoggerID: utils.String(d.Get("api_management_logger_id").(string)),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, apiName, diagnosticId, parameters, ""); err != nil {
+		return fmt.Errorf("creating or updating Diagnostic %q (Resource Group %q / API Management Service %q / API %q): %+v", diagnosticId, resourceGroup, serviceName, apiName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, serviceName, apiName, diagnosticId)
+	if err != nil {
+		return fmt.Errorf("retrieving Diagnostic %q (Resource Group %q / API Management Service %q / API %q): %+v", diagnosticId, resourceGroup, serviceName, apiName, err)
+	}
+	if resp.ID == nil {
```

Comment:
Can we also check the ID value is not empty?
```suggestion
	if resp.ID == nil || *resp.ID == "" {
```